src/test/java/HomeTest.java

```diff
@@ -0,0 +1,121 @@
+        providePassword("te$t$tudent");
+        loginToKoel();
+        //choosePlaylist;
+        choosePlaylist("Green Song");
+        //displayAllSongs;
+        displayAllSongs();
+        //Assertion;
+        Assert.assertTrue(getPlaylistDetails().contains(String.valueOf(countSongs())));
+
+    }
+
+    public String getPlaylistDetails() {
+        return driver.findElement(By.cssSelector("span.meta.text-secondary span.meta")).getText();
+
+    }
+
+    public void displayAllSongs() {
+        List<WebElement> songList = driver.findElements(By.cssSelector("section#playlistWrapper td.title"));
+        System.out.println("Number of Songs found:" + countSongs());
+        for (WebElement e : songList) {
+            System.out.println(e.getText());
+
+        }
+    }
+
+    public int countSongs() {
+        return driver.findElements(By.cssSelector("section#playlistWrapper td.title")).size();
+
+
+    }
+
+    public void choosePlaylist(String playListName) {
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//a[contains(text(),'" + playListName + "')]"))).click();
+
+    }
+
+    @Test
+    public void renamePlaylist() throws InterruptedException {
+
+        String updatedPlaylistSuccessMsg = "Updated playlist \"Sample Playlist.\"";
+
+        LoginPage loginPage = new LoginPage(driver);
+        HomePage homePage = new HomePage(driver);
+
+        loginPage.login();
+        homePage.doubleClickPlaylist();
+        homePage.enterNewName();
+        //Assertions;
+        Assert.assertEquals(homePage.getRenamePlaylistSuccessMsg(), updatedPlaylistSuccessMsg);
+
+    }
+    @Test
+    public void addSongPlayList() {
+        LoginPage loginPage = new LoginPage(driver);
+        HomePage homePage = new HomePage(driver);
+
+        loginPage.login();
+        homePage.choosePlaylist();
+        homePage.clickAddToBtn();
+        homePage.selectFirstSong();
+        homePage.clickViewAllBtn();
+        //homePage.searchSong();
+        homePage.getSongAddedSuccessMessage();
+        Assert.assertTrue(homePage.getSongAddedSuccessMessage());
+    }
+    }
+
+   /* public String getRenamePlaylistSuccessMsg(){
+        WebElement notification =wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("div.success.show")));
+        return notification.getText();
+    }
+
+    /*public void enterNewName() {
+        WebElement playlistInputField = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("[name='name']")));
+        playlistInputField.sendKeys(Keys.chord(Keys.CONTROL,"A",Keys.BACK_SPACE));
+        playlistInputField.sendKeys(newPlaylistName);
+        playlistInputField.sendKeys(Keys.ENTER);
+    }
+
+   /* public void doubleClickPlaylist() {
+        WebElement playlistElement = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector(".playlist:nth-child(3)")));
+        actions.doubleClick(playlistElement).perform();
```

Comment:
I recommend not using code comments: for, example, when you have a code that cannot be understood, then comments are not the solution. Try to rewrite the code or rename fields and other elements in a way that the reader will understand the action that you are doing there.